### DISCUSSION TITLE
Use Online Meeting Database post info outside meetings list

### DISCUSSION THREAD

**Discussion by landolfi-us** [code4recovery/12-step-meeting-list#1178] [General]
Hi,

First of all, huge thanks to all who built and maintain the TSML plugin. I'm sure it has saved many lives over its relatively brief history and will save many more.

I'm sure this question has been asked before but I can't find it here: Is there any way to use the online meeting database outside of the meeting guide? I know we could export it, but we'd prefer read access to the data in real time if possible. What we'd like to be able to do is use the meeting information to trigger submitting meeting information changes to the area registrar when a meeting is updated, but combine the online database info with additional user information about the meeting (such as number of group attendees) that is not in the online meeting database. Is this possible?

**Comment by kiyote33** [comment]
Hi @landolfi-us,
Besides exporting your meeting data in a CSV file format, you can also import the JSON feed into a SQL database or pull the data from a cached json file in the /wp-contents folder.  The meeting data stored in your WordPress mySql tables are available if your thinking of doing the development work yourself, but I think your question is asking us if we provide that access through our plugin. The answer of course is no, we don't have an interface built to access the data through the plugin but you can probably access it through your C Panel MySQL database access. I hope this answers your question...

**Reply by joshreisner** [reply]
> read access to the data

to expand on what @kiyote33 said above - you can provide another application with a read-only JSON view of your meetings by setting your sharing to open and copying the "Public Data Source" link

<img width="510" alt="Screenshot 2023-07-29 at 10 55 52 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/287e8f0e-8c60-45b9-95c7-0ae8dd92b6e6">

not sure if this fits what you're needing to do or not

**Reply by landolfi-us** [reply]
Thanks for the replies! I'm in the process of getting a crash course in JSON parsing and well on my way.

**Reply by landolfi-us** [reply]
This is a long shot, but is there any way to get the JSON feed to include everything that's in the export. in particular the meeting contact info (Contact1 to Contact3)?

**Reply by joshreisner** [reply]
if you set your Contact Visibility to "Public" on the settings page, then these contacts will show up both on meeting detail pages and in the JSON feed. this is not the default setting, of course!

<img width="452" alt="Screenshot 2023-07-30 at 1 18 59 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/3a2cc126-28ff-467e-836e-7769b46acb7c">

**Comment by kiyote33** [comment]
Hi again @landolfi-us,
If you're familiar with AA's stance on anonymity, I'm sure you will appreciate the can of worms you're digging into. A user in an admin role downloading potentially compromising information for internal use by a Central Office or other Recovery entity is one thing, but exposing that kind of data (contact name, phone, email) publicly via a json feed is not something we can contemplate. That being said, I would imagine a coder could pull all meeting data from the mySQL database and generate JSON output.  Hope this helps...

**Reply by landolfi-us** [reply]
Was just about to say I see why it's designed as it is and you're absolutely right that I don't want to get into this.

**Comment by landolfi-us** [comment]
A follow-up question and part of the reason for wanting to use the online meeting database--has there been any discussion of adding the GSO Fellowship Connection group number to the online meeting information? Right now I maintain that separately but it would be nice to have all the meeting info in one place. I was considering using the Group Notes field for this, and then re-importing with that field updated. If I did import a version of the export with group number added in that Group Notes field, would that data be preserved in future exports?

**Reply by tim-rohrer** [reply]
Hi--

That question does come up from time-to-time, but thus far we've opted to keep TSML focused on meeting data for the newcomers and others trying to find meetings.

As you've mentioned it is possible today for admins to enter the data you wish in the group notes field, yes that should persist if you export/import the data set.

**Reply by landolfi-us** [reply]
I screwed up our permalinks once before, I think with a reinstall of TSML, so I want to be extra careful. If I just import the same sheet I exported with data added to it in an existing data field, it won't affect our online meetings list except to replace all the records with my updated ones, right? In other words, a re-import is all I have to do? My biggest worry is it would change either permalinks or post IDs.

**Reply by joshreisner** [reply]
it should not affect the URLs of the meetings*, but i'm happy to try it out for you on our demo site. your plan is to export to CSV, edit the groups in a spreadsheet, and reimport?

* i think you mean post "slugs" aka the meeting url, the actual underlying wordpress post IDs will be changed, but this would not be noticed by regular users

**Reply by joshreisner** [reply]
@landolfi-us here's a short demo video - process seemed to work well on our demo site

https://www.loom.com/share/e90cac9b7770422a85cfa94b6500b456

**Reply by landolfi-us** [reply]
That is very kind of you to offer and yes, if you would be willing!

Actually, I was using the post ID to link my Formidable Forms entries to their meeting counterparts in the online meeting database, but with group IDs in the online meeting database I don't need to do that any more anyway.

[meetings_exported 7.28.23_with Groups in Group Notes.csv](https://github.com/code4recovery/12-step-meeting-list/files/12223645/meetings_exported.7.28.23_with.Groups.in.Group.Notes.csv)

**Reply by joshreisner** [reply]
Ok! Seemed like it worked pretty well. https://www.loom.com/share/1d8067fbda3b44f6947519af476c57ec

I noticed in the video that you already have a number of IDs in the group notes, so all you need to do for those meetings is add a group name.

One KEY THING to note is that Group Names are unique, so groups that share a name will also end up sharing group notes and contact details, so keep that in mind when naming your groups. You may need to add a suffix or two to keep different groups with the same name distinct

**Reply by landolfi-us** [reply]
Thanks very much! Could I also just use the Group field instead of both Group and Group Notes? I had not wanted to use that, thinking it was used for some other internal purpose like when you check the Contact Information checkbox "Part of a group".

**Reply by joshreisner** [reply]
The Group field (what i've been calling "group name" above) should be fine.
